### DISCUSSION TITLE
"Deeply Nested Models" example from the fastapi tutorials results in `422 Unprocessable Entity`

### DISCUSSION THREAD

**Discussion by mattangus** [fastapi/fastapi#6496] [Questions] [answered]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
# copied directly from https://fastapi.tiangolo.com/tutorial/body-nested-models/#deeply-nested-models
from typing import List, Optional, Set

from fastapi import FastAPI
from pydantic import BaseModel, HttpUrl

app = FastAPI()


class Image(BaseModel):
    url: HttpUrl
    name: str


class Item(BaseModel):
    name: str
    description: Optional[str] = None
    price: float
    tax: Optional[float] = None
    tags: Set[str] = set()
    images: Optional[List[Image]] = None


class Offer(BaseModel):
    name: str
    description: Optional[str] = None
    price: float
    items: List[Item]


@app.post("/offers/")
async def create_offer(offer: Offer):
    return offer
```

### Description

I am trying out the tutorial on Nested models and I keep getting 422 errors. Here are the commands run:

```
$ uvicorn webapp:app &
[1] 272890
$ INFO:     Started server process [272890]
INFO:     Waiting for application startup.
INFO:     Application startup complete.
INFO:     Uvicorn running on http://127.0.0.1:8000 (Press CTRL+C to quit)

$ curl -X 'POST' \
>   'http://127.0.0.1:8000/offers/' \
>   -H 'accept: application/json' \
>   -H 'Content-Type: application/json' \
>   -d '{
>   "name": "string",
>   "description": "string",
>   "price": 0,
>   "items": [
>     {
>       "name": "string",
>       "description": "string",
>       "price": 0,
>       "tax": 0,
>       "tags": [],
>       "images": [
>         {
>           "url": "string",
>           "name": "string"
>         }
>       ]
>     }
>   ]
> }'
INFO:     127.0.0.1:50456 - "POST /offers/ HTTP/1.1" 422 Unprocessable Entity
{"detail":[{"loc":["body","items",0,"images",0,"url"],"msg":"invalid or missing URL scheme","type":"value_error.url.scheme"}]}
```

The curl command was copied directly from `http://127.0.0.1:8000/docs`.

What am I doing wrong here? I've just been copy and pasting from the tutorial.

### Operating System

Linux

### FastAPI Version

0.72.0

### Python Version

3.8.10

**Comment by mattangus** [comment]
Don't mind me. figured it out :facepalm:

**Comment by tares003** [comment]
> Don't mind me. figured it out 🤦

I am also having the same issue, how did you solve it ? 

Thanks

**Comment by mattangus** [comment, marked as answer]
The key bit I was missing is this:

```
url: HttpUrl
```

Which means this line of the curl request was invalid

```
"url": "string",
```

I just needed to change it to a valid url like:

```
"url": "www.web.com", 
```
